cli/src/semgrep/dependency_aware_rule.py

```diff
@@ -109,9 +117,25 @@ def generate_unreachable_sca_findings(
     return non_reachable_matches, dep_rule_errors, targeted_lockfiles
 
 
+@lru_cache(maxsize=1000)
+def transivite_dep_is_also_direct(
+    package: str, deps: Tuple[Tuple[str, Transitivity], ...]
+) -> bool:
+    """
+    Assumes that [dep] is transitive
+    Checks if there is a direct version of the transitive dependency [dep]
+    """
+    for other_package, transitivity in deps:
+        if other_package == package and transitivity == Transitivity(Direct()):
+            return True
+    return False
```

Comment:
```suggestion
    return (package, Transitivity(Direct())) in deps
```

this might be a simpler equivalent